File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/BinaryTree/flatten_114.java
package algo.tzashinorpu.SpecialSubject.BinaryTree;

import algo.tzashinorpu.SpecialSubject.TreeNode;

public class flatten_114 {
    public void flatten(TreeNode root) {
        if (root == null) return;
        flatten(root.left);
        flatten(root.right);
        TreeNode left = root.left;
        TreeNode right = root.right;
        root.left = null;
        root.right = left;
        TreeNode p = root;
        while (p.right != null) p = p.right;
        p.right = right;
    }
}

File: Clion/ZeroTrac/Tree/flatten_114.cpp
//
// Created by TzashiNorpu on 10/25/2022.
//
#include "tree.h"

using namespace Tree;

void Solution::flatten(TreeNode *root) {
  if (!root) return;
  flatten(root->left);
  flatten(root->right);
  TreeNode *l = root->left;
  TreeNode *r = root->right;
  root->left = nullptr;
  root->right = l;
  TreeNode *p = root;
  while (p->right) p = p->right;
  p->right = r;
}